### PR TITLE
Object permissioning flexibility

### PR DESCRIPTION
**Pull request by piercetrey-figure**
* Allow any type of owner, as well as data access keys on scope to fetch
* Allow additional permissioned keys when putting an object (audience on dime as well as permissions in table)

### CODE REVIEW

File: server/src/main/kotlin/tech/figure/objectstore/gateway/service/ObjectService.kt [hunk 1]
```diff
@@ -35,14 +35,18 @@ class ObjectService(
...
+
         return objectStoreClient.osClient.put(
             ByteArrayInputStream(objectBytes),
             masterKey.publicKey,
             masterKey.signer(),
             objectSize,
-            setOf(requesterPublicKey),
+            additionalAudiences,
         ).get().hash.toByteArray().base64String().also { hash ->
-            objectPermissionsRepository.addAccessPermission(hash, requesterAddress, objectSize)
```
Comment (hyperschwartz) [inline server/src/main/kotlin/tech/figure/objectstore/gateway/service/ObjectService.kt:45]: I think we still need this line to ensure the creator can still fetch their stuff

Comment (hyperschwartz) [reply server/src/main/kotlin/tech/figure/objectstore/gateway/service/ObjectService.kt:45]: However, I'd think at least one test would blow up if the sender couldn't fetch their object anymore.  Maybe our testing around these functions is too sparse?

Comment (piercetrey-figure) [reply server/src/main/kotlin/tech/figure/objectstore/gateway/service/ObjectService.kt:45]: the `additionalAudiences` set does contain the requester's key, so they still get access permission. The verification blocks that verify `objectPermissionsRepository.addAccessPermission(objectHash, ownerKey.getAddress(false), objectBytes.size.toLong())` test that

Comment (piercetrey-figure) [reply server/src/main/kotlin/tech/figure/objectstore/gateway/service/ObjectService.kt:45]: maybe I should rename that value, as it is a little confusing that `additionalAudiences` contains the requester's key + `additionalAudienceKeys`... it is the 'additional audience' from the perspective of object store, but slightly more than that from this perspective

Comment (piercetrey-figure) [reply server/src/main/kotlin/tech/figure/objectstore/gateway/service/ObjectService.kt:45]: done been renam'd

Comment (hyperschwartz) [reply server/src/main/kotlin/tech/figure/objectstore/gateway/service/ObjectService.kt:45]: Whoops, my bad for not noticing that.  LGTM!

**Review by hyperschwartz — APPROVED** [review]
This is all super legit!

**Pull request merged**